### PR TITLE
✨ Add snooze notifications banner

### PR DESCRIPTION
**Pull request by astroash**
**Changes:**
- readd deprecate service naming as no longer toggling
- make call to new service on notifications page
- display banner dependant on snoozeUntil

### CODE REVIEW

File: server/routes/notification.js [hunk 1]
```diff
@@ -114,9 +116,10 @@ router.get(
       Promise.all([
         // This should be replaced with a non-paged call that shows only recent notifications.
         // This isn't an audit!!
-        notificationService.getShiftNotifications(req.user.username),
-        notificationService.getShiftNotificationsPaged(req.user.username, offset, perPage),
-      ]).then(([count, rows]) => {
+        DEPRECATEnotificationService.getShiftNotifications(req.user.username),
+        DEPRECATEnotificationService.getShiftNotificationsPaged(req.user.username, offset, perPage),
+        notificationService.getPreferences(req.user.token),
```
Comment (jonhindle) [inline server/routes/notification.js:121]: opportunity here to de-struct the vars we're using out of the `req` object. I appreciate this is more lift and shift so very much optional.

File: server/routes/notification.js [hunk 2]
```diff
@@ -114,9 +116,10 @@ router.get(
...
+      ]).then(([count, rows, preferences]) => {
```
Comment (jonhindle) [inline server/routes/notification.js:122]: destruct out the param you need `]).then(([count, rows, { snoozeUntil }]) => {`

File: server/routes/notification.js [hunk 3]
```diff
@@ -138,13 +141,15 @@ router.get(
           csrfToken: res.locals.csrfToken,
           hmppsAuthMFAUser: req.hmppsAuthMFAUser,
           authUrl: req.authUrl,
+          isSnoozed: moment(preferences.snoozeUntil).isAfter(moment()),
+          snoozeUntil: moment(preferences.snoozeUntil).format('dddd, Do MMMM, YYYY'),
```
Comment (jonhindle) [inline server/routes/notification.js:145]: Surely we don't need two values here? `isSnoozed` is implied by `snoozeUntil`.
You could extract this out to a utility and test separately then if you wanted then?
*pseudo code alert* 😬
```
const getTimeUntil = date => {
  const testMoment = moment(date)
  return testMoment.isAfter(moment()) ? 
    testMoment.format('dddd, Do MMMM, YYYY') : ''
}
```

File: server/routes/notification.js (hunk 2 above)
Comment (astroash) [reply server/routes/notification.js:122]: I left this on the `preferences` object as I use the property name `snoozeUntil` to pass to ejs as the formatted copy. I could change the ejs key `snoozeUntil` to `formattedSnoozeUntil` if you prefer, this is probably easier to read and avoids my sneaky/bad poor naming

Comment (jonhindle) [reply server/routes/notification.js:119]: You could also assign a new name in the deconstruct `]).then(([count, rows, { snoozeUntil: rawSnoozeUntil }]) => {`
I prefer externalising as a utility though. Its got a nice encapsulation about it?

**Pull request merged**